client/src/shapes/types/schema.ts

```diff
@@ -3,61 +3,63 @@ import type {
   GradientStop,
   Stroke as StrokeDefinition,
   TextArea as TextAreaDefinition,
-} from "./utility";
+} from './utility';
 
 export type AnchorPoint = {
   /**
    * the point on a 2d canvas this item is placed
    */
-  at: Coordinate
-}
+  at: Coordinate;
+};
 
 export type LineWidth = {
   /**
    * the visual thickness of the shape (in pixels)
    */
-  lineWidth?: number
-}
+  lineWidth?: number;
+};
 
 export type Stroke = {
   /**
    * the border around the shape
    */
-  stroke?: StrokeDefinition
-}
+  stroke?: StrokeDefinition;
+};
 
 export type TextArea = {
   /**
    * the text of the shape
    */
-  textArea?: TextAreaDefinition
-}
+  textArea?: TextAreaDefinition;
+};
 
 export type FillColor = {
   /**
    * the background color of the shape
    */
   fillColor?: string;
-}
+};
 
 export type Rotation = {
   /**
    * the rotation of the shape (in radians)
    */
-  rotation?: number
+  rotation?: number;
 };
 
 export type BorderRadius = {
   /**
    * the roundness of the shape's corners (in pixels)
+   * can be a single value or an array for each corner
+   * e.g. [topLeft, topRight, bottomRight, bottomLeft]
    */
-  borderRadius?: number;
-}
+  borderRadius?: number | [number, number, number, number];
```

Comment:
remember the trick to name the types of array indices? `[topLeft: number, ....]`

Reply:
I also may consider breaking this down further by creating a utility type called `BorderRadiusArrayValue` or something similar to DashPattern for a more atomic system